tests/test_qmark.py

```diff
@@ -55,32 +55,26 @@ def test_qmark_bulk_insert(db_parameters, run_v20_sqlalchemy):
     snowflake.connector.paramstyle = "qmark"
 
     engine = _get_engine_with_qmark(db_parameters)
-    con = engine.connect()
     import pandas as pd
 
-    try:
-        with con.begin():
-            con.exec_driver_sql(
-                """
-                create or replace table src(c1 int, c2 string) as select seq8(),
-                randstr(100, random()) from table(generator(rowcount=>100000))
-                """
-            )
-            con.exec_driver_sql(
-                """
-                create or replace table dst like src
-                """
-            )
-
-        with con.begin():
-            for data in pd.read_sql_query(
-                text("select * from src"), con, chunksize=16000
-            ):
-                data.to_sql(
-                    "dst", con, if_exists="append", index=False, index_label=None
+    with engine.connect() as con:
+        try:
+            with con.begin():
+                con.exec_driver_sql(
+                    """
+                    create or replace table src(c1 int, c2 string) as select seq8(),
+                    randstr(100, random()) from table(generator(rowcount=>100000))
+                    """
                 )
-
-    finally:
-        con.close()
-        engine.dispose()
-        snowflake.connector.paramstyle = "pyformat"
+                con.exec_driver_sql("create or replace table dst like src")
+
+                for data in pd.read_sql_query(
+                    text("select * from src"), con, chunksize=16000
+                ):
+                    data.to_sql(
+                        "dst", con, if_exists="append", index=False, index_label=None
+                    )
+
+        finally:
+            engine.dispose()
```

Comment:
We're inside of `engine`'s context manager, is `dispose` necessary here?

Reply:
`dispose` will shutdown the connection pool, so I feel we could keep it here showing the usage pattern to users